src/Adapters.Tests/Shared/ExceptionAssert.cs

```diff
@@ -0,0 +1,30 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="EventSourceTelemetryModuleTests.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.ApplicationInsights.Tests
+{
+    using System;
+    using VisualStudio.TestTools.UnitTesting;
+
+    public static class ExceptionAssert
+    {
+        public static void Throws<TException>(Action action, Action<TException> notifyException = null) where TException: Exception
```

Comment:
Unfortunately there isn't. MS test framework only has an attribute that you can apply to the whole test method, indicating that the method is supposed to throw, but this is not granular enough IMO. I want to be able to indicate which specific action is going to throw.

E.g. see discussion under http://stackoverflow.com/questions/933613/how-do-i-use-assert-to-verify-that-an-exception-has-been-thrown 

What I have done is patterned on NUint, which has this kind of assert in the core library.